metricflow-semantics/metricflow_semantics/naming/linkable_spec_name.py

```diff
@@ -24,9 +26,36 @@ class StructuredLinkableSpecName:
 
     entity_link_names: Tuple[str, ...]
     element_name: str
-    time_granularity: Optional[TimeGranularity] = None
+    time_granularity_name: Optional[str] = None
     date_part: Optional[DatePart] = None
 
+    def __post_init__(self) -> None:
+        """Post init method to ensure all name values are lower-case.
+
+        This implies that all values in the TimeGranularity enumeration must be lower-case.
+        """
+        assert self.element_name.lower() == self.element_name, (
+            f"Element name {self.element_name} is not all lower-case. This should have been addressed at model "
+            "parse time!"
+        )
+        assert all([link_name.lower() == link_name for link_name in self.entity_link_names]), (
+            f"Found entity link name that includes upper-case letters. Entity link names: {self.entity_link_names}. "
```

Comment:
Is this enforced already? That entity names must be lowercase?

Reply:
In theory, [yes](https://github.com/dbt-labs/dbt-semantic-interfaces/blob/main/dbt_semantic_interfaces/validations/unique_valid_name.py#L69-L84).

Reply:
Excellent! And a helpful link pointing me to how to add this validation for custom granularities too 🙃